backend/src/controllers/index.ts

```diff
@@ -0,0 +1,14 @@
+import { Request, Response } from "express";
+import { saveMessage, getUrl } from "../services";
+
+export const save = async (req: Request, res: Response) => {
+  const message = req.body.data;
+  const result = await saveMessage(message);
+  res.json(result.url);
+};
+
+export const get = async (req: Request, res: Response) => {
+  const id = req.params.id;
```

Comment:
we should check if id is present or not. If not, we can avoid the call to service right?